rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/service/JNDIResourceService.java

```diff
@@ -26,11 +26,11 @@ public JNDIResourceService(GraphContext context)
     }
 
     /**
-     * Create unique; if existing convert an existing {@link DataSourceModel} if one exists.
+     * Create unique; if existing convert an existing {@link JNDIResourceModel} if one exists.
      */
     public synchronized JNDIResourceModel createUnique(Set<ProjectModel> applications, String jndiName)
     {
-        JNDIResourceModel jndiResourceModel = getUniqueByProperty(DataSourceModel.JNDI_LOCATION, jndiName);
+        JNDIResourceModel jndiResourceModel = getUniqueByProperty(JNDIResourceModel.JNDI_LOCATION, jndiName);
         if (jndiResourceModel == null)
```

Comment:
I think this might have been intentional. Also I think we agreed to have interface HasJndi. Can be done later.